src/frontend/components/SelectContent/index.tsx

```diff
@@ -0,0 +1,354 @@
+import { Nullable } from '../../utils/types';
+
+const messages = defineMessages({
+  title: {
+    defaultMessage: 'Select LTI content',
+    description: `Title for the LTI content selector, where the LTI Consumer can see the available video and documents
+      , and will be able to select them.`,
+    id: 'components.SelectContent.title',
+  },
+  addVideo: {
+    defaultMessage: 'Add a video',
+    description: `Text displayed on a button to add a new video.`,
+    id: 'components.SelectContent.addVideo',
+  },
+  addDocument: {
+    defaultMessage: 'Add a document',
+    description: `Text displayed on a button to add a new document.`,
+    id: 'components.SelectContent.addDocument',
+  },
+  cancel: {
+    defaultMessage: 'Cancel',
+    description: `Text displayed on a button to cancel content selection.`,
+    id: 'components.SelectContent.cancel',
+  },
+  confirm: {
+    defaultMessage: 'Select',
+    description: `Text displayed on a button to confirm content selection.`,
+    id: 'components.SelectContent.confirm',
+  },
+  newVideo: {
+    defaultMessage: 'New video',
+    description: `Default LTI consumer title for a new video.`,
+    id: 'components.SelectContent.newVideo',
+  },
+  newDocument: {
+    defaultMessage: 'New document',
+    description: `Default LTI consumer title for a new document.`,
+    id: 'components.SelectContent.newDocument',
+  },
+  uploaded: {
+    defaultMessage: 'Uploaded',
+    description: `Text helper displayed if a video or a document is uploaded.`,
+    id: 'components.SelectContent.uploaded',
+  },
+  notUploaded: {
+    defaultMessage: 'Not uploaded',
+    description: `Text helper displayed if a video or a document is not uploaded.`,
+    id: 'components.SelectContent.notUploaded',
+  },
+  readyToShow: {
+    defaultMessage: 'Ready to show',
+    description: `Text helper displayed if a video or a document is ready to show to students.`,
+    id: 'components.SelectContent.readyToShow',
+  },
+  notReadyToShow: {
+    defaultMessage: 'Not ready to show',
+    description: `Text helper displayed if a video or a document is not ready to show to students.`,
+    id: 'components.SelectContent.notReadyToShow',
+  },
+  select: {
+    defaultMessage: 'Select {content_title}',
+    description: `Title used for a video or a document select.`,
+    id: 'components.SelectContent.select',
+  },
+  titleEdit: {
+    defaultMessage: 'LTI Content title',
+    description: `Label for LTI title content input.`,
+    id: 'components.SelectContent.titleEdit',
+  },
+});
+
+const IconBox = styled.span`
+  font-size: 70px;
+  text-align: center;
+  padding: 40px;
+`;
+
+const IconStatus = ({
+  message,
+  GrommetIcon,
+  color,
+}: {
+  message: string;
+  GrommetIcon: Icon;
+  color: string;
+}) => (
+  <Box direction="row" gap="small" pad="small">
+    <GrommetIcon a11yTitle={message} color={color} />
+    <Text>{message}</Text>
+  </Box>
+);
+
+const AssertedIconStatus = ({
```

Comment:
I don't know if this component is really useful. Using directly the `IconStatus` is enough IMO

Reply:
I think it might be useful for showing binary states of objects.
But if you think it's overkill, I'll remove it.